lib/pool.js

```diff
@@ -59,17 +59,15 @@ class Pool extends EventEmitter {
         this.dispatch(item.opts, item.handler)
       }
 
-      if (queue.isEmpty()) {
-        if (pool[kNeedDrain]) {
-          pool[kNeedDrain] = false
-          pool.emit('drain')
-        }
+      if (pool[kNeedDrain] && !this.busy) {
+        pool[kNeedDrain] = false
+        pool.emit('drain')
+      }
```

Comment:
why this if was removed?

Reply:
I don't understand the question.

Reply:
you removed the `if (queue.isEmpty())` check. Why?

Reply:
I replaced it with `this.busy` which contains the same check through `this[kPending] > 0`

Reply:
the empty check is moved to just `if (pool[kClosedResolve] && queue.isEmpty()) {`